### PR TITLE
Support augmentation for classification task: cutmix, mixup

### PR DESCRIPTION
**Pull request by illian01**
## Description

Please include a summary in English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #215

You should link at least one existing issue for PR. Before your create a PR, please check to see if there is an issue for this change.  
PRs from forked repository not accepted.

## Change(s)

- Add `cutmix` and `mixup` augmentation for classification task
- Since `cutmix` and `mixup` needs different samples, they are applied in `collate_fn`
- To distinguish with existing transform functions, `mixup` and `cutmix` added as `mix_transform`

## Changelog

If you PR to `dev` branch, please add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file and include a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`.

### CODE REVIEW

File: src/netspresso_trainer/dataloaders/builder.py [hunk 1]
```diff
@@ -100,7 +103,24 @@ def build_dataset(conf_data, conf_augmentation, task: str, model_name: str):
 def build_dataloader(conf, task: str, model_name: str, train_dataset, eval_dataset, profile=False):
 
     if task == 'classification':
-        collate_fn = None
+        if hasattr(conf.augmentation, 'mix_transforms'):
```
Comment (illian01) [inline src/netspresso_trainer/dataloaders/builder.py:106]: @deepkyu 
I added `mixup` and `cutmix` as `mix_transforms`.
For consistency, how about change `recipe` in augmentation config to `transforms`?

Comment (deepkyu) [reply src/netspresso_trainer/dataloaders/builder.py:106]: Agreed. Sounds better.

File: src/netspresso_trainer/dataloaders/augmentation/custom.py [hunk 2]
```diff
@@ -349,6 +351,172 @@ def __repr__(self):
         return format_string
 
 
+class RandomMixup:
```
Comment (deepkyu) [inline src/netspresso_trainer/dataloaders/augmentation/custom.py:354]: Class attribute `visualize` is needed.
If it is not supporting the debug with visualizer, set the value as `False`.

File: src/netspresso_trainer/dataloaders/augmentation/custom.py [hunk 3]
```diff
@@ -349,6 +351,172 @@ def __repr__(self):
...
+
+        return mixup_samples, mixup_targets
+
+    def __repr__(self) -> str:
+        return "{}(num_classes={}, p={}, alpha={}, inplace={})".format(
+            self.__class__.__name__, self.num_classes, self.p, self.alpha, self.inplace
+        )
+
+
+class RandomCutmix:
```
Comment (deepkyu) [inline src/netspresso_trainer/dataloaders/augmentation/custom.py:430]: Same here.

Class attribute `visualize` is needed.
If it is not supporting the debug with visualizer, set the value as `False`.

**Review by deepkyu — CHANGES_REQUESTED** [review]

**Review by deepkyu — COMMENTED** [review]
Nice.

Hey @illian01, could you update the default classification augmentation config including `cutmix`?

**Review by deepkyu — APPROVED** [review]

**Pull request merged**